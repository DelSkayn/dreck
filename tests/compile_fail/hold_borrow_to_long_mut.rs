use dreck::*;

fn main() {
    let mut owner = unsafe{ Owner::new(marker::Invariant::new())};
    let a = {
        let root = unsafe { Root::new(&owner) };
        let a = root.add(0);
        a.borrow_mut_untraced(&mut owner)
    };
    assert_eq!(*a, 0);
}
