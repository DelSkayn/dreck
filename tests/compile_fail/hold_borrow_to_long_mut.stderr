error[E0597]: `root` does not live long enough
 --> tests/compile_fail/hold_borrow_to_long_mut.rs:7:17
  |
7 |         let a = root.add(0);
  |                 ^^^^^^^^^^^ borrowed value does not live long enough
8 |         a.borrow_mut_untraced(&mut owner)
  |         --------------------------------- borrow later used here
9 |     };
  |     - `root` dropped here while still borrowed
